src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,2091 @@
+
+@add_start_docstrings(
+    "The bare PatchTST Model outputting raw hidden-states without any specific head.",
+    PATCHTST_START_DOCSTRING,
+)
+class PatchTSTModel(PatchTSTPreTrainedModel):
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__(config)
+
+        if config.scaling == "mean" or config.scaling is True:
+            self.scaler = PatchTSTMeanScaler(dim=1, keepdim=True)
+        elif config.scaling == "std":
+            self.scaler = PatchTSTStdScaler(dim=1, keepdim=True)
+        else:
+            self.scaler = PatchTSTNOPScaler(dim=1, keepdim=True)
+
+        self.patching = PatchTSTPatchify(
+            config.context_length,
+            patch_length=config.patch_length,
+            stride=config.stride,
+        )
+        self.mask_input = config.mask_input
+
+        if self.mask_input:
+            self.masking = PatchTSTMasking(
+                mask_type=config.mask_type,
+                mask_ratio=config.mask_ratio,
+                mask_patches=config.mask_patches,
+                mask_patch_ratios=config.mask_patch_ratios,
+                channel_consistent_masking=config.channel_consistent_masking,
+                unmasked_channel_indices=config.unmasked_channel_indices,
+                mask_value=config.mask_value,
+                seed_number=config.seed_number,
+            )
+        else:
+            self.masking = nn.Identity()
+        self.encoder = PatchTSTEncoder(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def forward(
+        self,
+        past_values: torch.Tensor,
+        past_observed_mask: Optional[torch.Tensor] = None,
+        future_values: Optional[torch.Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, PatchTSTModelOutputWithNoAttention]:
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if past_observed_mask is None:
+            past_observed_mask = torch.ones_like(past_values)
+
+        # x: tensor [bs x sequence_length x num_input_channels]
+        scaled_past_values, loc, scale = self.scaler(past_values, past_observed_mask)
+
+        # patched_values: [bs x num_input_channels x num_patches x patch_length] for pretrain
+        patched_values = self.patching(scaled_past_values)
+        if self.mask_input:
+            masked_values, mask = self.masking(patched_values)
+        else:
+            masked_values, mask = self.masking(patched_values), None
+        encoder_output = self.encoder(masked_values, output_hidden_states=output_hidden_states)
+
+        hidden_states = encoder_output.last_hidden_state
+        encoder_states = encoder_output.hidden_states
+
+        if not return_dict:
+            return tuple(v for v in [hidden_states, encoder_states, patched_values, mask, loc, scale] if v is not None)
+        return PatchTSTModelOutputWithNoAttention(
```

Comment:
can we leverage the same output classes as the existing time series models? This way we can have the same API for all time series models (have pipeline support in the future etc).